lib/postrank-uri.rb

```diff
@@ -235,5 +235,12 @@ def valid?(uri)
 
       is_valid
     end
+
+    def squeeze_slashes(path)
+      path = path.squeeze('/')
+      return path unless path =~ (/https?:\/[^\/]/)
+      path = path.gsub(/https:\/+/, 'https://')
+      path.gsub(/http:\/+/, 'http://')
+    end
```

Comment:
Seems like we could simplify this a bit and make it faster. 

Instead of running squeeze and then doing a regex check, followed by gsub's.. Run a gsub that finds all double-slash occurrences that are not preceded by `http:` or `https:` and collapse those. In short, I think we can replace the call to squeeze with a single gsub? WDYT?